packages/astro/src/core/render/README.md

```diff
@@ -0,0 +1,70 @@
+# `core/render`
+
+This directory contains most of Astro's high-level rendering APIs, including `renderPage`, `createRenderContext`, etc.
+
+- For rendering an Astro file, see [src/runtime/server/](../../runtime/server/).
+- For rendering an endpoint, see [src/core/endpoint/](../endpoint/).
+
+## Concepts
+
+The codebase has a few abstractions for rendering:
+
+### `RenderContext`
+
+Each render (page or endpoint) requires a `RenderContext` that contains:
+
+- The `Request` object
+- The matched `route`
+- The matched `pathname` (without the `base`)
+- The route `params` and `props`
+- Additional `styles`, `links`, `scripts`
+- And more!
+
+The `RenderContext` is agnostic to what's being rendered. 
+
+**Permitted state:** `RenderContext` should only contain per-request information.
+
+### `Environment`
+
+Every app (dev and prod) has one `Environment`. It contains a subset of the Astro `settings`, `config`, and `routes` information needed for Astro's runtime to work.
+
+In dev, all of `settings`, `config`, and `routes` are available, so the `Environment` (aka `DevelopmentEnvironment`) is derived directly from them. In prod, the `Environment` is derived from the `SSRManifest`, an intermediate layer that helps keep the build output lean.
+
+**Permitted state:** `Environment` should only contain the global state shared across all requests.
+
+### `SSRManifest`
+
+An `SSRManifest` is created during a build to save information needed to create an `Environment` during runtime start-up. The values should be serializable (`buildManifest`) and deserializable (`deserializeManifest`).
```

Comment:
Considering this documentation, we should rename this function to `serializeManifest`. What do you think? (not in this PR)

Reply:
Yeah we should definitely do that. Especially that the serialization part is quite spreaded, making both side-by-side would be good long term.